src/game/collide.c

```diff
@@ -114,19 +135,32 @@ void Collide_GetCollisionInfo(
     coll->front_floor = height;
     coll->front_ceiling = ceiling;
     coll->front_type = g_HeightType;
-    if (coll->slopes_are_walls && coll->front_type == HT_BIG_SLOPE
-        && coll->front_floor < 0) {
-        coll->front_floor = -32767;
-    } else if (
-        coll->slopes_are_pits && coll->front_type == HT_BIG_SLOPE
-        && coll->front_floor > 0) {
-        coll->front_floor = 512;
-    } else if (
-        coll->lava_is_pit && coll->front_floor > 0 && g_TriggerIndex
-        && (g_TriggerIndex[0] & DATA_TYPE) == FT_LAVA) {
-        coll->front_floor = 512;
+    skip_slope_check = false;
+    walkable_heights = Room_GetWalkableHeights(floor, x, ypos, z);
+    for (int32_t i = 0; i < walkable_heights->count; i++) {
+        int32_t item_height = (int32_t)walkable_heights->items[i];
```

Comment:
`const int32_t` (but only in the variable type, not in the cast, as making primitive casts const serves no purpose)